File: 2023/1/rust/src/part1.rs
// Advent of Code 2023 - Day 1 - Part 1

static FILE: &str = "../input.txt";

use std::fs::File;
use std::io::BufRead;

pub fn part1() {
    let mut sum = 0;

    let f = File::open(FILE).unwrap();
    let reader = std::io::BufReader::new(f);
    for line in reader.lines() {
        let numbers = get_numbers(&line.unwrap());
        if numbers.len() < 1 {
            continue;
        } else {
            sum += format!("{}{}", numbers[0], numbers[numbers.len() - 1]).parse::<i32>().unwrap();
        }
    }
    println!("SUM: {}", sum);
}

fn get_numbers(line: &str) -> Vec<i32> {
    let mut numbers = Vec::<i32>::new();
    for char in line.chars() {
        if char.is_digit(10) {
            numbers.push(char.to_digit(10).unwrap() as i32);
        }
    }
    return numbers;
}

File: 2023/1/part_1.py
def get_numbers(line):
    numbers_in_line = []
    for char in line:
        if char.isdigit():
            numbers_in_line.append(char)
    return numbers_in_line


sum_numbers = 0
with open("input.txt", "r") as f:
    for line in f.readlines():
        print(line.__str__().strip())
        numbers = get_numbers(line)
        if len(numbers) < 1:
            continue
        else:
            sum_numbers += int("{}{}".format(numbers[0], numbers[-1]))
print(sum_numbers.__str__())
